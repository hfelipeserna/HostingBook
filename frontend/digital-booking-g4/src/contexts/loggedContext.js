import { createContext } from "react";

const loggedContext = createContext();

export default loggedContext;