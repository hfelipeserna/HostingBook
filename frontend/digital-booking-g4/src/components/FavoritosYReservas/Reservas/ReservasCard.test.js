import React from "react";
import { BrowserRouter, Switch, Route, Redirect } from "react-router-dom";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import { MemoryRouter } from 'react-router-dom';
import loggedContext from '../../../contexts/loggedContext'
import ReservasCard from './ReservaCard'
import Router from "react-router-dom";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createSerializer } from "enzyme-to-json";
import { configure, mount } from "enzyme";

describe(" Reservas card Reserva test", function () {
    const alojamiento = {

        id: 1,
        nombre: "meliá",
        categoria: {
            id: 1,
            titulo: "Hoteles",
            descripcion: "807.105 hoteles",
            urlImagen: "https://proyectointegradorimagenes2.s3.us-east-2.amazonaws.com/Categorias/Hoteles.jfif"
        },
        ciudad: {
            id: 2,
            nombre: "La Plata",
            pais: "Argentina",
            latitud: -34.9214,
            longitud: -57.9544
        },
        imagenes: [{
            id: 2,
            imagenTitulo: "Baño,",
            imagenUrl: "https://proyectointegradorimagenes2.s3.us-east-2.amazonaws.com/Imagenes/banio.jpg"
        }, {
            id: 7,
            imagenTitulo: "Principal",
            imagenUrl: "https://proyectointegradorimagenes2.s3.us-east-2.amazonaws.com/Alojamientos/Producto1.jpg"
        }],

        caracteristicas: [{
            id: 2,
            nombre: "Piscina",
            icono: "fas fa-swimmer"
        }],
        direccion: "123"

    }
    const reserva = {
        id: 1,
        horaEntrada: 2,
        nombre: "este Es el nombre de la reserva",
        apellido: "Este es el apellido de la reserva",
        mail: "este es el mail de la reserva",
        fechaIngreso: "22-03-2022",
        fechaEgreso: "22-03-2022"
    }
    beforeAll(() => {
        jest.mock('react-router-dom', () => ({
            ...jest.requireActual('react-router-dom'),
            useHistory: () => ({
                push: jest.fn(),
                goBack: jest.fn()
            }),
        }));
        global.localStorage = {
            i2x_token: 'someToken',
            getItem: function () {
                return 'someToken'
            }
        };
        localStorage.setItem("id", 1)

    })

    test("Renderiza Default", () => {
        const renderizacionSimple = render(
            <BrowserRouter>
                <Switch>
                    <Route exact path="/" >
                        <loggedContext.Provider value={{ isLogged: true }}>
                            <ReservasCard alojamiento={alojamiento} reserva={reserva}></ReservasCard>
                        </loggedContext.Provider>
                    </Route>
                </Switch>
            </BrowserRouter>,
            { wrapper: MemoryRouter })
        expect(renderizacionSimple).toMatchSnapshot()

    })
})