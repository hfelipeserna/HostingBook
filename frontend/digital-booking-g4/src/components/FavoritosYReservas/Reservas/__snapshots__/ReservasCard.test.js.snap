// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Reservas card Reserva test Renderiza Default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tarjetaAlojamiento"
        data-aos="fade-up"
      >
        <div
          class="imagenAlojamiento"
          style="background-image: url(https://proyectointegradorimagenes2.s3.us-east-2.amazonaws.com/Alojamientos/Producto1.jpg);"
        />
        <div
          class="descripcionAlojamiento"
        >
          <div
            class="informacionPrincipal"
          >
            <div
              class="nombreAlojamiento"
            >
              <div
                class="tipoYCalificacion"
              >
                <h4>
                  Hoteles
                </h4>
                <span
                  aria-label="Estrellas"
                  class="estrellas"
                >
                  <i
                    class="fas fa-star estrellaVacia"
                  />
                  <i
                    class="fas fa-star estrellaVacia"
                  />
                  <i
                    class="fas fa-star estrellaVacia"
                  />
                  <i
                    class="fas fa-star estrellaVacia"
                  />
                  <i
                    class="fas fa-star estrellaVacia"
                  />
                </span>
              </div>
              <h2>
                meliá
              </h2>
            </div>
            <div
              class="puntajeAlojamiento"
            >
              
              <div
                class="detalle"
              >
                Sin calificaciones
              </div>
            </div>
          </div>
          <div
            class="informacionDetalle"
          >
            <div
              class="ubicacion"
            >
              <i
                class="fas fa-map-marker-alt"
              />
              <p>
                La Plata
                <span
                  class="mostrarEnMapa"
                >
                  Mostrar en el mapa
                </span>
              </p>
            </div>
            <div
              class="servicios"
            >
              <i
                class="fas fa-swimmer"
              />
            </div>
          </div>
          <div
            class="reservaDetalleCard"
          >
            <h3>
              Datos de la reserva
            </h3>
            <div>
              <ul
                class="reservaDetalleCardLeft"
              >
                <li>
                  Número de reserva: 
                  <span>
                    1
                  </span>
                </li>
                <li>
                  Nombre: 
                  <span>
                    este Es el nombre de la reserva
                  </span>
                </li>
                <li>
                  Apellido: 
                  <span>
                    Este es el apellido de la reserva
                  </span>
                </li>
                <li>
                  Mail: 
                  <span>
                    este es el mail de la reserva
                  </span>
                </li>
              </ul>
              <ul
                class="reservaDetalleCardRight"
              >
                <li>
                  Horario de llegada: 
                  <span>
                    2
                    hs
                  </span>
                </li>
                <li>
                  Check in: 
                  <span>
                    2022/03/22
                  </span>
                </li>
                <li>
                  Check out: 
                  <span>
                    2022/03/22
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="reservaButtonsContainer"
          >
            <input
              class="filledButton reservaBtnCalificar"
              type="submit"
              value="Calificar"
            />
            <div
              class="reservaButtonsContainerVerMasCancelar"
            >
              <a
                href="/product/1/features"
              >
                <input
                  class="filledButton reservaBtnVerMas"
                  type="submit"
                  value="Ver más"
                />
              </a>
              <input
                class="outlinedButton reservaBtnCancelar"
                type="submit"
                value="Cancelar reserva"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tarjetaAlojamiento"
      data-aos="fade-up"
    >
      <div
        class="imagenAlojamiento"
        style="background-image: url(https://proyectointegradorimagenes2.s3.us-east-2.amazonaws.com/Alojamientos/Producto1.jpg);"
      />
      <div
        class="descripcionAlojamiento"
      >
        <div
          class="informacionPrincipal"
        >
          <div
            class="nombreAlojamiento"
          >
            <div
              class="tipoYCalificacion"
            >
              <h4>
                Hoteles
              </h4>
              <span
                aria-label="Estrellas"
                class="estrellas"
              >
                <i
                  class="fas fa-star estrellaVacia"
                />
                <i
                  class="fas fa-star estrellaVacia"
                />
                <i
                  class="fas fa-star estrellaVacia"
                />
                <i
                  class="fas fa-star estrellaVacia"
                />
                <i
                  class="fas fa-star estrellaVacia"
                />
              </span>
            </div>
            <h2>
              meliá
            </h2>
          </div>
          <div
            class="puntajeAlojamiento"
          >
            
            <div
              class="detalle"
            >
              Sin calificaciones
            </div>
          </div>
        </div>
        <div
          class="informacionDetalle"
        >
          <div
            class="ubicacion"
          >
            <i
              class="fas fa-map-marker-alt"
            />
            <p>
              La Plata
              <span
                class="mostrarEnMapa"
              >
                Mostrar en el mapa
              </span>
            </p>
          </div>
          <div
            class="servicios"
          >
            <i
              class="fas fa-swimmer"
            />
          </div>
        </div>
        <div
          class="reservaDetalleCard"
        >
          <h3>
            Datos de la reserva
          </h3>
          <div>
            <ul
              class="reservaDetalleCardLeft"
            >
              <li>
                Número de reserva: 
                <span>
                  1
                </span>
              </li>
              <li>
                Nombre: 
                <span>
                  este Es el nombre de la reserva
                </span>
              </li>
              <li>
                Apellido: 
                <span>
                  Este es el apellido de la reserva
                </span>
              </li>
              <li>
                Mail: 
                <span>
                  este es el mail de la reserva
                </span>
              </li>
            </ul>
            <ul
              class="reservaDetalleCardRight"
            >
              <li>
                Horario de llegada: 
                <span>
                  2
                  hs
                </span>
              </li>
              <li>
                Check in: 
                <span>
                  2022/03/22
                </span>
              </li>
              <li>
                Check out: 
                <span>
                  2022/03/22
                </span>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="reservaButtonsContainer"
        >
          <input
            class="filledButton reservaBtnCalificar"
            type="submit"
            value="Calificar"
          />
          <div
            class="reservaButtonsContainerVerMasCancelar"
          >
            <a
              href="/product/1/features"
            >
              <input
                class="filledButton reservaBtnVerMas"
                type="submit"
                value="Ver más"
              />
            </a>
            <input
              class="outlinedButton reservaBtnCancelar"
              type="submit"
              value="Cancelar reserva"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
